litestar/config/app.py

```diff
@@ -82,7 +83,7 @@ class AppConfig:
     """A sync or async function called after the response has been awaited. It receives the
     :class:`Request <.connection.Request>` object and should not return any values.
     """
-    allowed_hosts: list[str] | AllowedHostsConfig | None = field(default=None)
+    allowed_hosts: AllowedHostsConfig | None = field(default=None)
```

Comment:
This is a breaking change. Users might want to pass a list directly into the config and plugins may receive this config object and expect a `list[str]` here, which they can append to

Reply:
Ah I thought `AppConfig` was internal (or at least not supposed to be instantiated by the user). 
Reverted.

Reply:
You'll usually instantiate it directly when using `Litestar.from_config`

Reply:
Also, `AppConfig` is very public API for plugin authors.